youtube-video-element.js

```diff
@@ -0,0 +1,542 @@
+// https://developers.google.com/youtube/iframe_api_reference
+
+const EMBED_BASE = 'https://www.youtube.com/embed';
+const API_URL = 'https://www.youtube.com/iframe_api';
+const API_GLOBAL = 'YT';
+const API_GLOBAL_READY = 'onYouTubeIframeAPIReady';
+const MATCH_SRC =
+  /(?:youtu\.be\/|youtube\.com\/(?:embed\/|v\/|watch\?v=|watch\?.+&v=))((\w|-){11})/;
+
+const templateShadowDOM = document.createElement('template');
+templateShadowDOM.innerHTML = `
+<style>
+  :host {
+    display: inline-block;
+    line-height: 0;
+    position: relative;
+    width: 300px;
+    height: 150px;
+  }
+  iframe {
+    position: absolute;
+    top: 0;
+    left: 0;
```

Comment:
Alternatively, use [`inset`](https://developer.mozilla.org/en-US/docs/Web/CSS/inset) which allows setting all 4 props at once
```suggestion
    inset: 0;
```

Reply:
I'll keep it in mind. 
the iframe still has a 100% width and height which in some cases was preferred over right: 0 and bottom: 0. I forget the exact details around it.